backend/inspirehep/accounts/decorators.py

```diff
@@ -34,6 +36,8 @@ def wrapped_function(*args, **kwargs):
                 abort(401)
 
             if roles:
+                # superuser can do everything
+                roles.append(Roles.superuser.value)
```

Comment:
What about
```
if not roles:
    roles = []
# Rest of the function body without `ìf`
```
so you can decorate with `@login_required_with_roles()` to allow only `superuser`?

Reply:
No, you have to specify at least 1 role. I didn't change behavior when you are not specifying any role as this would break something which we are maybe not testing ;) Better safe than sorry.